lib/validatorCompiler.js

```diff
@@ -0,0 +1,71 @@
+const Ajv = require('ajv')
+const ajvOpenApi = require('ajv-openapi')
+
+const openAjv = (compileObj, data) => {
+  const ajvOptions = {
+    schemaId: 'auto',
+    format: 'full',
+    coerceTypes: true,
+    unknownFormats: 'ignore',
+    useDefaults: true,
+    nullable: true
+  }
+
+  const openApiOptions = {
+    useDraft04: true
+  }
+
+  const ajv = ajvOpenApi(
+    new Ajv(ajvOptions),
+    openApiOptions
+  )
+  return ajv.compile(compileObj)(data)
+}
+
+const binaryValidation = (data) => {
+  return openAjv({ format: 'binary', type: 'string' }, data)
```

Comment:
I means to take their validation as example and implement our own.

Reply:
ok, waitting for next weekend to work on that